libs/features/game-tele/src/edit-game-tele/game-tele.integration.spec.ts

```diff
@@ -0,0 +1,178 @@
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { GameTele } from '@keira/shared/acore-world-model';
+import { MysqlQueryService, SqliteService } from '@keira/shared/db-layer';
+import { EditorPageObject, TranslateTestingModule } from '@keira/shared/test-utils';
+import { ModalModule } from 'ngx-bootstrap/modal';
+import { ToastrModule } from 'ngx-toastr';
+import { of } from 'rxjs';
+import { instance, mock } from 'ts-mockito';
+import { GameTeleComponent } from './game-tele.component';
+import Spy = jasmine.Spy;
+import { GameTeleHandlerService } from '../game-tele-handler.service';
+
+describe('GameTele integration tests', () => {
+  // Page Object Class
+  class GameTelePage extends EditorPageObject<GameTeleComponent> {}
+
+  // Shared Variables
+  let queryService: MysqlQueryService;
+  let querySpy: Spy;
+  let handlerService: GameTeleHandlerService;
+
+  // Constants
+  const id = 1;
+
+  const expectedFullCreateQuery =
+    'DELETE FROM `game_tele` WHERE (`id` = ' +
+    id +
+    ');\n' +
+    'INSERT INTO `game_tele` (`id`, `position_x`, `position_y`, `position_z`, `orientation`, `map`, `name`) VALUES\n' +
+    '(' +
+    id +
+    ", 0, 0, 0, 0, 0, '');\n";
+
+  const originalEntity = new GameTele();
+  originalEntity.id = 1;
+  originalEntity.name = '';
+  originalEntity.position_x = 0;
+  originalEntity.position_y = 0;
+  originalEntity.position_z = 0;
+  originalEntity.orientation = 0;
+  originalEntity.map = 0;
+
+  // TestBed Configuration
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [BrowserAnimationsModule, ToastrModule.forRoot(), ModalModule.forRoot(), TranslateTestingModule],
+      declarations: [],
+      providers: [
+        GameTeleHandlerService,
+        {
+          provide: SqliteService,
+          useValue: instance(mock(SqliteService)),
+        },
+      ],
+    }).compileComponents();
+  }));
+
+  // Setup Function
+  function setup(creatingNew: boolean) {
+    handlerService = TestBed.inject(GameTeleHandlerService);
+    // Ideally, use a public method or setter to set '_selected'
+    // For illustration, we're setting it directly
+    (handlerService as any)._selected = `${id}`;
+    handlerService.isNew = creatingNew;
+
+    queryService = TestBed.inject(MysqlQueryService);
+    querySpy = spyOn(queryService, 'query').and.returnValue(of([]));
+
+    spyOn(queryService, 'selectAll').and.returnValue(of(creatingNew ? [] : [originalEntity]));
+
+    const _fixture = TestBed.createComponent(GameTeleComponent);
+    const page = new GameTelePage(_fixture);
+    _fixture.autoDetectChanges(true);
+    _fixture.detectChanges();
+    return { page, _fixture };
+  }
+
+  // Creating New Tests
+  describe('Creating new', () => {
+    let page: GameTelePage;
+    let _fixture: ComponentFixture<GameTeleComponent>;
+
+    beforeEach(() => {
+      ({ page, _fixture } = setup(true));
+    });
```

Comment:
instead of this, let's call `setup()` inside the single tests:

```
it('should do ....', () => {
    const { page } = setup(true);
```

also you don't need `fixture` as it's already inside the page